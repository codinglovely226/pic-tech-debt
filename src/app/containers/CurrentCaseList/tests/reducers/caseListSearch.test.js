/* eslint-disable */

import caseListSearchReducer from '../../reducers/caseListSearch';

import {
  updateCaseListSearchCallerLocation,
  updateCaseListSearchCallerOrganisation,
  updateCaseListSearchCentreName,
  updateCaseListSearchFilter,
  updateCaseListSearchName,
  updateCaseListSearchPostcode,
  updateCaseListSearchSex,
  updateCaseListSearchTime,
} from '../../actions';

describe('caseListSearchReducer', () => {
  it('returns the initial state', () => {
    expect(caseListSearchReducer(undefined, {})).toMatchSnapshot();
  });
  it('update caller location', () => {
    expect(caseListSearchReducer({}, updateCaseListSearchCallerLocation('Home'))).toMatchSnapshot();
  });
  it('update caller organisation', () => {
    expect(caseListSearchReducer({}, updateCaseListSearchCallerOrganisation('Hospital'))).toMatchSnapshot();
  });
  it('update centre name', () => {
    expect(caseListSearchReducer({}, updateCaseListSearchCentreName('Panadol'))).toMatchSnapshot();
  });
  it('update filter', () => {
    expect(caseListSearchReducer({}, updateCaseListSearchFilter('all'))).toMatchSnapshot();
  });
  it('update name', () => {
    expect(caseListSearchReducer({}, updateCaseListSearchName('Smith'))).toMatchSnapshot();
  });
  it('update postcode', () => {
    expect(caseListSearchReducer({}, updateCaseListSearchPostcode(8000))).toMatchSnapshot();
  });
  it('update sex', () => {
    expect(caseListSearchReducer({}, updateCaseListSearchSex('male'))).toMatchSnapshot();
  });
  it('update time', () => {
    expect(caseListSearchReducer({}, updateCaseListSearchTime('2018'))).toMatchSnapshot();
  });
});
