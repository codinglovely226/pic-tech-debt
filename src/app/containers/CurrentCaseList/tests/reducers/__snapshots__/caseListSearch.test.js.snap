exports[`caseListSearchReducer returns the initial state 1`] = `
Object {
  "callerLocation": "",
  "callerOrganisation": "",
  "centreName": "",
  "filter": "open",
  "name": "",
  "naturalId": "",
  "postcode": "",
  "sex": "",
  "time": "",
}
`;

exports[`caseListSearchReducer update caller location 1`] = `
Object {
  "callerLocation": "Home",
  "callerOrganisation": "",
  "centreName": "",
  "filter": "open",
  "name": "",
  "naturalId": "",
  "postcode": "",
  "sex": "",
  "time": "",
}
`;

exports[`caseListSearchReducer update caller organisation 1`] = `
Object {
  "callerLocation": "",
  "callerOrganisation": "Hospital",
  "centreName": "",
  "filter": "open",
  "name": "",
  "naturalId": "",
  "postcode": "",
  "sex": "",
  "time": "",
}
`;

exports[`caseListSearchReducer update centre name 1`] = `
Object {
  "callerLocation": "",
  "callerOrganisation": "",
  "centreName": "Panadol",
  "filter": "open",
  "name": "",
  "naturalId": "",
  "postcode": "",
  "sex": "",
  "time": "",
}
`;

exports[`caseListSearchReducer update filter 1`] = `
Object {
  "callerLocation": "",
  "callerOrganisation": "",
  "centreName": "",
  "filter": "all",
  "name": "",
  "naturalId": "",
  "postcode": "",
  "sex": "",
  "time": "",
}
`;

exports[`caseListSearchReducer update name 1`] = `
Object {
  "callerLocation": "",
  "callerOrganisation": "",
  "centreName": "",
  "filter": "open",
  "name": "Smith",
  "naturalId": "",
  "postcode": "",
  "sex": "",
  "time": "",
}
`;

exports[`caseListSearchReducer update postcode 1`] = `
Object {
  "callerLocation": "",
  "callerOrganisation": "",
  "centreName": "",
  "filter": "open",
  "name": "",
  "naturalId": "",
  "postcode": 8000,
  "sex": "",
  "time": "",
}
`;

exports[`caseListSearchReducer update sex 1`] = `
Object {
  "callerLocation": "",
  "callerOrganisation": "",
  "centreName": "",
  "filter": "open",
  "name": "",
  "naturalId": "",
  "postcode": "",
  "sex": "male",
  "time": "",
}
`;

exports[`caseListSearchReducer update time 1`] = `
Object {
  "callerLocation": "",
  "callerOrganisation": "",
  "centreName": "",
  "filter": "open",
  "name": "",
  "naturalId": "",
  "postcode": "",
  "sex": "",
  "time": "2018",
}
`;
